python/ray/data/_internal/stats.py

```diff
@@ -1182,11 +1182,12 @@ def to_string(self) -> str:
                 )
             if self.total_time.get():
                 out += "* Total time overall: {}\n".format(fmt(self.total_time.get()))
-            out += "* Num blocks local: {}\n".format(self.iter_blocks_local)
-            out += "* Num blocks remote: {}\n".format(self.iter_blocks_remote)
-            out += "* Num blocks unknown location: {}\n".format(
-                self.iter_unknown_location
-            )
+            if DataContext.get_current().enable_get_object_locations_for_metrics:
```

Comment:
While we're at it, this is a good time to clean this formatting up as well.

How about we create another indented block here for "Block locations", and we can move it to before or after the other two blocks, which are all about CPU time spent instead of data movement.